### PR TITLE
Rotate image in Imviz

### LINKED ISSUE #1299: Imviz coordinates display shows wrong X and Y for dithered image on load
**Reporter: Pey-Lian Lim**

Originally reported by [~cpacifici] at https://github.com/spacetelescope/jdaviz/pull/1293#issuecomment-1118120046 .

Workflow to reproduce:

1. Load data in Imviz example notebook.
2. Link them by WCS.
3. Zoom in to object of interest.
4. Mouseover it. Note the X and Y.
5. Blink.
6. Note the X and Y do not change. This is wrong. It should have changed, which implies X and Y from Step 4 were actually pixel coordinates for the reference image.
7. Blink again. X and Y now disagree with those in Step 4.

 --- 

 DISCLAIMER: This issue was autocreated by the Jdaviz Issue Creation Bot on behalf of the reporter. If any information is incorrect, please contact [Duy Nguyen](https://github.com/duytnguyendtn)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to allow image rotation in Imviz.

Still need ~a lot of~ some work but here is what I get with the stuff in the concept notebook added in this PR:

Screenshot 2022-06-07 172839

Bonus: Compass zoom box now can rotate too.

Screenshot 2022-06-08 155529

### TODO

- [x] ~How to make Compass play nice with rotation?~ We will just disable the zoom box in Compass when this plugin is active or something.
- [x] How to hide the reference WCS but keep it as reference?
- [x] How will this work when multiple images are loaded?
- [x] How will this work with GWCS?
- [x] How to undo/redo rotation?
- [x] Performance considerations.
- [x] Manually test a variety of other workflows: Multiple viewers, etc.
- [x] When POC works, implement in plugin.
- [x] Clean up concept notebook.
- [x] Bug: Coordinates info bar can get out of sync after rotating with 3 images in the concept notebook. Maybe my Lab is messed up? Try in notebook and see.
    - Fix #1299
- [x] Take out the stuff moved to #1392
- [ ] Write unit tests.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-1826)

### CODE REVIEW

**Comment by astrofrog** [comment]
The approach of using a dummy dataset with a rotated WCS is 'what I would do too given the current glue infrastructure. I think what we ideally want to be able to do is to have the ability to have such datasets have a special status in glue which would make them 'hidden' from most lists except for the reference data in the image viewer, but that's not possible for now. But it is something I could prioritize if desired.

**Comment by pllim** [comment]
> But it is something I could prioritize if desired.

Would it be much easier to do this in Glue? I was about to do some real hacky stuff in Jdaviz to hide it. Should I not go down that rabbit hole, @astrofrog ?

File: jdaviz/app.py [hunk 1]
```diff
@@ -1017,12 +1017,19 @@ def _viewer_by_reference(self, reference):
...
             The viewer class instance.
         """
         viewer_item = self._viewer_item_by_reference(reference)
 
-        return self._viewer_store[viewer_item['id']]
+        # There is a degeneracy between reference and ID.
+        # Imviz falls back to ID for user-created viewers.
+        if viewer_item is not None:
+            viewer = self._viewer_store[viewer_item['id']]
+        else:  # Maybe it is ID
```
Comment (pllim) [inline jdaviz/app.py:1029]: I used to have this in the app somewhere but maybe it was removed in one of the refactoring. @kecnry , I know your pet peeve is the reference vs ID stuff, so I want to make sure you are okay with this change.

**Comment by pllim** [comment]
Looks like this PR might be rejected, so I am going to take out the bug fixes and open new PR for that.

**Pull request closed without merging**